src/server/components/dashboard/controllers.ts

```diff
@@ -549,11 +537,143 @@ export async function listItemsDeleteController(
   }
 }
 
+export async function listItemsEditGetController(
+  req: Request,
+  res: Response
+): Promise<void> {
+  const { listId, listItemId } = req.params;
+  const userId = req.user?.userData.id;
+
+  if (userId === undefined) {
+    return res.redirect(authRoutes.logout);
+  }
+
+  const list = await findListById(listId);
+  const listItem = await findListItemById(listItemId);
+
+  if (list === undefined) {
+    res.status(404).send({
+      error: {
+        message: `Could not find list ${listId}`,
+      },
+    });
+  } else if (listItem === undefined) {
+    res.status(404).send({
+      error: {
+        message: `Could not find list item ${listItemId}`,
+      },
+    });
+  } else if (list.type !== listItem?.type) {
```

Comment:
when I try to use `list.type !== listItem!.type` it produces `ESLint: This assertion is unnecessary since it does not change the type of the expression`. I couldn't find a reference to the `!.` syntax - what is it meant to do?